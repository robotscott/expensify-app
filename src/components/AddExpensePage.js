import React from "react";
import { connect } from "react-redux";
import ExpenseForm from "./ExpenseForm";
import { startAddExpense } from "../actions/expenses";

export class AddExpensePage extends React.Component {
	onSubmit = expense => {
		this.props.startAddExpense(expense);
		this.props.history.push("/");
	};
	render() {
		return (
			<div>
				<div class="page-header">
					<div class="content-container">
						<h1 class="page-header__title">Add Expense</h1>
					</div>
				</div>
				<div class="content-container">
					<ExpenseForm onSubmit={this.onSubmit} />
				</div>
			</div>
		);
	}
}

const mapDispatchToProps = dispatch => ({
	startAddExpense: expense => dispatch(startAddExpense(expense))
});

export default connect(undefined, mapDispatchToProps)(AddExpensePage);
