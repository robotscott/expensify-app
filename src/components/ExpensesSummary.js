import React from "react";
import { connect } from "react-redux";
import selectExpenses from "../selectors/expenses";
import expenseTotal from "../selectors/expense-total";
import numeral from "numeral";

export const ExpensesSummary = props => {
	const expenseCount = props.expenses.length;
	const expenseWord = expenseCount === 1 ? "expense" : "expenses";
	const total = numeral(expenseTotal(props.expenses) / 100).format("$0,0.00");
	return (
		<div>
			<h1>
				Viewing {expenseCount} {expenseWord} totalling {total}.
			</h1>
		</div>
	);
};

const mapStateToProps = state => {
	return {
		expenses: selectExpenses(state.expenses, state.filters)
	};
};

export default connect(mapStateToProps)(ExpensesSummary);
