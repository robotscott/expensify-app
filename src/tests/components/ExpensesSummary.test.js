import React from "react";
import { shallow } from "enzyme";
import { ExpensesSummary } from "../../components/ExpensesSummary";
import expenses from "../fixtures/expenses";

test("should render ExpenseSummary correctly", () => {
	const wrapper = shallow(<ExpensesSummary expenses={expenses} />);
	expect(wrapper).toMatchSnapshot();
});

test("should render ExpenseSummary with 1 value", () => {
	const wrapper = shallow(<ExpensesSummary expenses={[expenses[0]]} />);
	expect(wrapper).toMatchSnapshot();
});