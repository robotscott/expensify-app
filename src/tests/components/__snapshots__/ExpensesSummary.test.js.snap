// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseSummary correctly 1`] = `
<div>
  <h1>
    Viewing 
    3
     
    expenses
     totalling 
    $1,141.95
    .
  </h1>
</div>
`;

exports[`should render ExpenseSummary with 1 value 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    $1.95
    .
  </h1>
</div>
`;
